### PR TITLE
clean up dev/tox/etc, strip out six (we only support 3.6 from now on)

### PR DESCRIPTION
**Pull request by drmorr0**
This commit strips out six from taskproc -- there's nothing that's using taskproc on Python2.7 and Python2.7 is basically EOLed anyways, so let's not bother.

I also clean up the development environment/tox/etc a little bit here.

### CODE REVIEW

File: examples/cluster/docker-compose.yaml [hunk 1]
```diff
@@ -43,7 +42,6 @@ services:
       - dynamodb
     volumes:
       - ../..:/src
-      - /var/run/docker.sock:/var/run/docker.sock
```
Comment (keymone) [inline examples/cluster/docker-compose.yaml]: why is this removed? iirc it's needed for playground to be able to use host's docker.

File: examples/cluster/playground/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM ubuntu:trusty
+FROM ubuntu:bionic
```
Comment (keymone) [inline examples/cluster/playground/Dockerfile]: this could cause to weird differences in behavior between example-cluster and what's running on stages/prod?

Comment (Rob-Johnson) [reply examples/cluster/playground/Dockerfile]: yeah, can we just use xenial?

File: requirements.txt [hunk 3]
```diff
@@ -1,5 +0,0 @@
-# Library so -e .
```
Comment (Rob-Johnson) [inline requirements.txt]: why dont we need this anymore?

File: examples/cluster/docker-compose.yaml (hunk 1 above)
Comment (drmorr0) [reply examples/cluster/docker-compose.yaml]: Apparently this is no longer allowed?  I get this error when trying to start the playground:

```
tarting cluster_zookeeper_1 ... done
Starting cluster_dynamodb_1  ... done
Starting cluster_mesosmaster_1 ... done
Starting cluster_mesosagent_1  ... done
ERROR: Cannot create container for service playground: b"You're not allowed to mount /run/docker.sock in rw mode.\nYou are only allowed to bind mount things from these directories: [/nail/home/drmorr:rw /nail/bulkdata:ro /nail/etc:ro /nail/etc/srv-configs:ro /nail/sys/bin/indocker:ro /nail/live/private:ro /nail/srv:ro /nail/home/dockerbuild:ro /nail/scratch/drmorr:rw /scratch/drmorr:rw /etc/boto_cfg:ro /etc/boto.cfg:ro /etc/mrjob.conf:ro /etc/mrjob.cpu-intensive.conf:ro /etc/mrjob.memory-intensive.conf:ro /etc/mrjob.minimal.conf:ro /etc/ssh/ssh_known_hosts:ro /etc/yelp_clog.json:ro /var/run/synapse/services/:ro /var/run/synapse/sockets/:ro /run/synapse/services/:ro /run/synapse/sockets/:ro /etc/nodebot.conf:ro /etc/localtime:ro /usr/share/zoneinfo/:ro /etc/passwd:ro /etc/group:ro /nail/etc/zookeeper_discovery/:ro /nail/etc/kafka_discovery/:ro /nail/tmp/:rw /nail/opt/:ro /etc/send_nsca.cfg:ro /var/run/fusebox/:ro /run/fusebox/:ro]"
```

Comment (Rob-Johnson) [reply examples/cluster/docker-compose.yaml]: thats a Yelp thing - you need to ``unset DOCKER_HOST``

File: requirements.txt (hunk 3 above)
Comment (drmorr0) [reply requirements.txt]: For libraries our recommended configuration is to put dependencies in setup.py and not have a requirements.txt:

> Libraries should not have deps = -rrequirements.txt in their tox files. Having a requirements.txt is not recommended in libraries.

https://yelpwiki.yelpcorp.com/display/WEBCORE/Applications+vs+Libraries

File: examples/cluster/docker-compose.yaml (hunk 1 above)
Comment (drmorr0) [reply examples/cluster/docker-compose.yaml]: aha thanks.
